frame/tx-pause/src/tests.rs

```diff
@@ -16,3 +16,173 @@
+		));
+	});
+}
+
+#[test]
+fn can_pause_system_call() {
+	new_test_ext().execute_with(|| {
+		let call = Call::System(frame_system::Call::remark { remark: vec![] });
+
+		assert_ok!(TxPause::pause_call(
+			Origin::signed(mock::PauseOrigin::get()),
+			b"System".to_vec().try_into().unwrap(),
+			b"remark".to_vec().try_into().unwrap(),
+		));
+
+		assert_err!(
+			call.clone().dispatch(Origin::signed(0)),
+			frame_system::Error::<Test>::CallFiltered
+		);
+	});
+}
+
+#[test]
+fn can_pause_specific_call() {
+	new_test_ext().execute_with(|| {
+		let call_paused = Call::Balances(pallet_balances::Call::transfer { dest: 1, value: 1 });
+		let call_not_paused =
+			Call::Balances(pallet_balances::Call::transfer_keep_alive { dest: 1, value: 1 });
+
+		assert_ok!(TxPause::pause_call(
+			Origin::signed(mock::PauseOrigin::get()),
+			b"Balances".to_vec().try_into().unwrap(),
+			b"transfer".to_vec().try_into().unwrap(),
+		));
+
+		assert_err!(
+			call_paused.clone().dispatch(Origin::signed(0)),
+			frame_system::Error::<Test>::CallFiltered
+		);
+		assert_ok!(call_not_paused.clone().dispatch(Origin::signed(0)));
+	});
+}
+
+#[test]
+fn can_unpause_specific_call() {
+	new_test_ext().execute_with(|| {
+		let call_paused = Call::Balances(pallet_balances::Call::transfer { dest: 1, value: 1 });
+
+		assert_ok!(TxPause::pause_call(
+			Origin::signed(mock::PauseOrigin::get()),
+			b"Balances".to_vec().try_into().unwrap(),
+			b"transfer".to_vec().try_into().unwrap(),
+		));
+		assert_err!(
+			call_paused.clone().dispatch(Origin::signed(0)),
+			frame_system::Error::<Test>::CallFiltered
+		);
+
+		assert_ok!(TxPause::unpause_call(
+			Origin::signed(mock::UnpauseOrigin::get()),
+			b"Balances".to_vec().try_into().unwrap(),
+			b"transfer".to_vec().try_into().unwrap(),
+		));
+		assert_ok!(call_paused.clone().dispatch(Origin::signed(0)));
+	});
+}
+
+// GENERAL FAIL/NEGATIVE TESTS ---------------------
+
+#[test]
+fn fails_to_pause_self() {
+	new_test_ext().execute_with(|| {
+		let within_bound_name =
+			(0u8..(MaxNameLen::get() - 1).try_into().unwrap()).collect::<Vec<_>>();
+
+		assert_noop!(
+			TxPause::pause_call(
+				Origin::signed(mock::PauseOrigin::get()),
+				b"TxPause".to_vec().try_into().unwrap(),
+				within_bound_name.try_into().expect("Function name under MaxNameLen")
+			),
+			Error::<Test>::IsUnpausable
+		);
+	});
+}
+
+#[test]
+fn fails_to_pause_unpausable_pallet() {
+	new_test_ext().execute_with(|| {
+		let within_bound_name =
+			(0u8..(MaxNameLen::get() - 1).try_into().unwrap()).collect::<Vec<_>>();
+
+		assert_noop!(
+			TxPause::pause_call(
+				Origin::signed(mock::PauseOrigin::get()),
+				b"UnpausablePallet".to_vec().try_into().unwrap(),
+				within_bound_name.try_into().expect("Function name under MaxNameLen")
+			),
+			Error::<Test>::IsUnpausable
+		);
+	});
+}
+
+#[test]
+#[should_panic(expected = "Pallet name under MaxNameLen")] // TODO - this test likely redundant/unneeded, as BoundedVec is effectively what we are testing.
```

Comment:
Yeah, this test does not make sense, also i do not know why you would test for panic here versus the returned `Err` from the result

Reply:
`too_long_name.try_into().expect("Pallet name under MaxNameLen")`
￼panics at the expect statement, This could be changed to  `too_long_name.try_into()?` and test for the error I think, but will remove this test anyhow as it's not needed/useful.